### DISCUSSION TITLE
Workflow for upstreaming patches

### DISCUSSION THREAD

**Discussion by Mic92** [jesseduffield/lazygit#1182] [Q&A]
I have my own branch where I have a collections of patches on my local master branch.
I would like to create a PR branch for each of them that is based on the upstream branch.

What is your workflow for that?
In the `Commits` window there is branch-off function bound to `n`. However this will base a branch based on current master with unrelated commits that also live in my local master.

**Edit**
My workflow in plain git would look like that (**origin** is my my fork, **upstream** is the project I want to upstream patches to)

```console
$ git checkout -b some-feature-branch-1 upstream/master
$ git cherry-pick some_commits_related_to_feature_1....
$ git push origin some-feature-branch-1 # than click on the PR link provided by github/gitlab in the push event)
$ git checkout -b some-feature-branch-2 upstream/master
$ git cherry-pick some_commits_related_to_feature_2....
$ git push origin some-feature-branch-2 # and open a different PR just with these commits
```

Usually I would actually use `tig master` while being in `some-feature-branch-1`  instead of `git cherry-pick` and have a cherry-pick key binding to quickly get my commits into a feature branch. ( I am still missing this command line flag in `lazygit` to see a different branch from the current one, but this is another story :))

**Comment by jesseduffield** [comment]
First off: apologies for the late reply, github didn't send me notifications about comments in these discussion threads!

I'm not sure that I understand your use case. Could you describe how you would go about doing this on the command line?

**Reply by Mic92** [reply]
Thanks. I expanded the explanation.
